common.rkt

```diff
@@ -140,23 +140,35 @@
 (define (reify tm st)
   (define index -1)
   (let ((results (let loop ((tm tm) (st st))
-              (define t (walk tm (state-sub st)))
-              (cond ((pair? t) (loop (cdr t) (loop (car t) st)))
-                    ((var? t)  (set! index (+ 1 index))
-                               (state (extend-sub t (reified-index index) (state-sub st)) (state-diseq st) (state-types st)))
-                    (else      st)))))
+                   (define t (walk tm (state-sub st)))
+                   (cond ((pair? t) (loop (cdr t) (loop (car t) st)))
+                         ((var? t)  (set! index (+ 1 index))
+                                    (state (extend-sub t (reified-index index) (state-sub st)) (state-diseq st) (state-types st)))
+                         (else      st)))))
     (let* ((walked-sub (walk* tm results))
+           (diseq (sort (state-diseq st) (lambda (x y) (< (length x) (length y)))))
+           (st (diseq-simplify (state (state-sub st) diseq (state-types st))))
+           (diseq (walk* (map pretty-diseq (state-diseq st)) results))
+           (diseq (filter fresh-var-diseq? diseq))
+           (diseq (if (null? diseq) '() (list (cons '=/= diseq))))
```

Comment:
Before this update to `diseq`, I would also sort each individual disjunction using `term<?` to normalize their order.